app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -794,39 +476,589 @@ public void onCharacteristicChanged(BluetoothGatt gatt, BluetoothGattCharacteris
+        for (i = 0; i < cTable.length; i++) {
+            if (cTable[i] == ch) break;
+        }
+        return i;
+    }
+
+    // Status for Watchface
+    public static boolean isRunning() {
+        return lastState.equals("Not Running") || lastState.startsWith("Stopping", 0) ? false : true;
+    }
+
+    public static DataMap getWatchStatus() {
+        DataMap dataMap = new DataMap();
+        dataMap.putString("lastState", lastState);
+        if (last_transmitter_Data != null)
+            dataMap.putLong("timestamp", last_transmitter_Data.timestamp);
+        dataMap.putInt("mStaticState", mStaticState);
+        dataMap.putInt("last_battery_level", last_battery_level);
+        dataMap.putLong("retry_time", retry_time);
+        dataMap.putLong("failover_time", failover_time);
+        dataMap.putString("static_last_hexdump", static_last_hexdump);
+        dataMap.putString("static_last_sent_hexdump", static_last_sent_hexdump);
+        return dataMap;
+    }
+
+    public static void setWatchStatus(DataMap dataMap) {
+        lastStateWatch = dataMap.getString("lastState", "");
+        last_transmitter_DataWatch = new TransmitterData();
+        last_transmitter_DataWatch.timestamp = dataMap.getLong("timestamp", 0);
+        mStaticStateWatch = dataMap.getInt("mStaticState", 0);
+        last_battery_level_watch = dataMap.getInt("last_battery_level", -1);
+        retry_time_watch = dataMap.getLong("retry_time", 0);
+        failover_time_watch = dataMap.getLong("failover_time", 0);
+        static_last_hexdump_watch = dataMap.getString("static_last_hexdump", "");
+        static_last_sent_hexdump_watch = dataMap.getString("static_last_sent_hexdump", "");
+    }
+
+    public static String getBestLimitterHardwareName() {
+        if (static_use_nrf && blueReader.isblueReader()) {
+            return "BlueReader";
+        } else if (static_use_nrf && Tomato.isTomato()) {
+            return xdrip.getAppContext().getString(R.string.tomato);
+        } else if (static_use_blukon) {
+            return xdrip.getAppContext().getString(R.string.blukon);
+        } else if (static_use_transmiter_pl_bluetooth) {
+            return "Transmiter PL";
+        } else if (static_use_rfduino_bluetooth) {
+            return "Rfduino";
+        } else return LIMITTER_NAME;
+    }
+
+    // data for MegaStatus
+    public static List<StatusItem> megaStatus() {
+        final List<StatusItem> l = new ArrayList<>();
+
+        final boolean forced_wear = Home.get_forced_wear();
+
+        l.add(new StatusItem("Phone Service State", lastState + (forced_wear ? " (Watch Forced)" : "")));
+
+        if (lastError != null) {
+            l.add(new StatusItem("Last Error", lastError, StatusItem.Highlight.NOTICE, "long-press", () -> lastError = null));
+        }
+
+        l.add(new StatusItem("Bluetooth Device", JoH.ucFirst(getStateStr(mStaticState))));
+
+        if (mStaticState == STATE_CONNECTING) {
```

Comment:
Here we show details about how long the current attempt to connect has been going on for